libs/coin-evm/docs/transaction.md

```diff
@@ -0,0 +1,31 @@
+# `transaction.ts`
+
+`transaction.ts` contains a set of helpers related to transactions in general. Some of these are standards, such as the serializers that are exported by default.
+
+## Contants
+
+##### DEFAULT_GAS_LIMIT
+The default value of `gasLimit` for a transaction. This represents the minimum amount of gas used by a coin transaction. 
+For Layer 1 coin transactions (transactions of coins without involving smart contracts), the `gasLimit` should not exceed this value. :tw-26a0: However, for Layer 2 transactions, it may not hold true as some Layer 2 solutions require payment for data availability. This means you have to pay for both L2 fees and the settlement of the rollup on L1, and this can be accomplished through gas or additional fees.
```

Comment:
Do we add this gas related to data availability on each transaction in current impl ? is it only related to the first tx on some L2 we currently support ?

Reply:
Data availability is paid on every transaction on L2. Depending on the L2, it's either paid by greatly increasing the gasLimit returned by the transaction simulation (like Arbitrum and Orbit chains) which means directly paying this in the transaction, or by retroactively withdrawing funds from the sender once the transaction is validated (like Optimism and the OP stack) which means it's not included in the transaction. You can still use an oracle to have a rough estimation of the price of the DA on OP chains. See: https://github.com/LedgerHQ/ledger-live/blob/develop/libs/coin-evm/src/api/node/rpc.common.ts#L245